src/library/base/man/options.Rd

```diff
@@ -832,6 +838,40 @@ if(tmp)
    cat("do cleanup\n")
 }
 
+myprint <- function(x, ...) {
```

Comment:
hmm I don't think it should be shielded from the global environment, because that's where users define methods for their classes.

If the custom print method is defined in a package, it is probably intentional that any print method defined there should be dispatched to. I assume that these methods would be registered anyway, and so would have global effects.

In general, print methods (or any S3 method for a foreign generic) should only be defined for classes that one owns. That applies to a package exporting an autoprint function, and that applies to the R user defining stuff in the global environment.

Reply:
Ah good point. If the autoprint function is exported from a package, it should normally inherit from its namespace and will then be shielded from such interferences in the global environment.

As for autoprint functions defined in RProfile, I'll adjust the examples to use the namespace, for the extra safety. Thanks!